tower-balance/src/p2c/service.rs

```diff
@@ -88,7 +98,7 @@ where
 
 impl<D, Req> Balance<D, Req>
 where
-    D: Discover,
+    D: Discover + Unpin,
```

Comment:
This makes me really sad, but it is required since we need to call `Discover::poll_discover` (which takes `Pin<&mut Self>`) from `Service::poll_ready` (which takes `&mut Self`).

Reply:
See also #319

Reply:
Yeah, this is fine 👍 can we make sure we have docs on why this is required maybe, just point people to that issue and what they can do to get Unpin (aka box the discover)?